### PR TITLE
Add metrics query APIs

### LINKED ISSUE #376: Add period metrics querying API
Add period metrics querying API

### PR DESCRIPTION
**Pull request by baixinsui**
fix #376

### CODE REVIEW

File: modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/monitor/Monitor.java [hunk 1]
```diff
@@ -11,67 +11,205 @@
...
+        if (Objects.isNull(from)) {
+            from = System.currentTimeMillis() - 60 * 1000;
+        }
+        if (Objects.isNull(to)) {
+            to = System.currentTimeMillis();
+        }
+        return new ServiceMetricRequest(deployResources, credential, monitorType, from, to, period);
+    }
+
+    private void checkParamsValid(Long from, Long to) {
```
Comment (swaroopar) [inline modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/monitor/Monitor.java:205]: method name can be -> validateToAndFromValues.

File: modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/monitor/Monitor.java [hunk 2]
```diff
@@ -11,67 +11,205 @@
...
+        if (Objects.nonNull(from) && Objects.nonNull(to)) {
```
Comment (swaroopar) [inline modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/monitor/Monitor.java:206]: i think it is also better to validate if the `to` and from `values` are also valid time stamps.

File: modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/monitor/Monitor.java [hunk 3]
```diff
@@ -11,67 +11,205 @@
...
+        }
+        return new ResourceMetricRequest(deployResource, credential, monitorType, from, to, period);
+    }
+
+    private ServiceMetricRequest getServiceMetricRequest(List<DeployResource> deployResources,
+                                                         CredentialDefinition credential,
+                                                         MonitorResourceType monitorType,
+                                                         Long from, Long to, Integer period) {
+        if (Objects.isNull(from)) {
+            from = System.currentTimeMillis() - 60 * 1000;
```
Comment (swaroopar) [inline modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/monitor/Monitor.java:197]: we must document this default value in the API also.

**Review by swaroopar — CHANGES_REQUESTED** [review]
comments added. Also there are merge conflicts. Please check that too.

**Review by swaroopar — CHANGES_REQUESTED** [review]
comments added. Also there are merge conflicts. Please check that too.

File: modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/monitor/Monitor.java (hunk 1 above)
Comment (baixinsui) [reply modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/monitor/Monitor.java:205]: Ok.

File: modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/monitor/Monitor.java (hunk 2 above)
Comment (baixinsui) [reply modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/monitor/Monitor.java:206]: Any number is a valid unix timestamps. I think it is not necessary. Or we can define a minimum start time '1970-01-01 00:00:00.000' with number 0?

File: modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/monitor/Monitor.java (hunk 3 above)
Comment (baixinsui) [reply modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/monitor/Monitor.java:197]: I will add it.

File: modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/monitor/Monitor.java [hunk 4]
```diff
@@ -11,67 +11,205 @@
...
+        }
+        if (Objects.isNull(to)) {
+            to = System.currentTimeMillis();
```
[the remark this answers is not in the document]
Comment (baixinsui) [reply modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/monitor/Monitor.java:200]: I will add it.

**Pull request closed without merging**